subprojects/logging/src/main/java/org/gradle/internal/logging/LoggingConfigurationBuildOptionFactory.java

```diff
@@ -106,8 +105,8 @@ private LogLevel parseLogLevel(String value) {
                 }
                 return logLevel;
             } catch (IllegalArgumentException e) {
-                String message = String.format("Value '%s' given for %s system property is invalid.  (must be one of quiet, warn, lifecycle, info, or debug)", value, gradleProperty);
-                throw new IllegalArgumentException(message, e);
+                Origin.GRADLE_PROPERTY.handleInvalidValue(this, value, "must be one of quiet, warn, lifecycle, info, or debug)");
+                return null;
```

Comment:
I think I'd prefer this pattern for the method which makes it more readable.

```
LogLevel logLevel = null;

try {
    // parse log level
} catch (IllegalArgumentException e) {
    // handle exception
}

return logLevel;

Reply:
Done.